src/streams/tls.h

```diff
@@ -20,4 +20,412 @@
+	{0xc0a9, "PSK-AES256-CCM8",                   "TLS_PSK_WITH_AES_256_CCM_8"},
+	{0xc0aa, "DHE-PSK-AES128-CCM8",               "TLS_PSK_DHE_WITH_AES_128_CCM_8"},
+	{0xc0ab, "DHE-PSK-AES256-CCM8",               "TLS_PSK_DHE_WITH_AES_256_CCM_8"},
+	{0xc0ac, "ECDHE-ECDSA-AES128-CCM",            "TLS_ECDHE_ECDSA_WITH_AES_128_CCM"},
+	{0xc0ad, "ECDHE-ECDSA-AES256-CCM",            "TLS_ECDHE_ECDSA_WITH_AES_256_CCM"},
+	{0xc0ae, "ECDHE-ECDSA-AES128-CCM8",           "TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8"},
+	{0xc0af, "ECDHE-ECDSA-AES256-CCM8",           "TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8"},
+	{0xcc13, "ECDHE-RSA-CHACHA20-POLY1305-OLD",   "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256_OLD"},
+	{0xcc14, "ECDHE-ECDSA-CHACHA20-POLY1305-OLD", "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256_OLD"},
+	{0xcc15, "DHE-RSA-CHACHA20-POLY1305-OLD",     "TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256_OLD"},
+	{0xcca8, "ECDHE-RSA-CHACHA20-POLY1305",       "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xcca9, "ECDHE-ECDSA-CHACHA20-POLY1305",     "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xccaa, "DHE-RSA-CHACHA20-POLY1305",         "TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xccab, "PSK-CHACHA20-POLY1305",             "TLS_PSK_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xccac, "ECDHE-PSK-CHACHA20-POLY1305",       "TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xccad, "DHE-PSK-CHACHA20-POLY1305",         "TLS_DHE_PSK_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xccae, "RSA-PSK-CHACHA20-POLY1305",         "TLS_RSA_PSK_WITH_CHACHA20_POLY1305_SHA256"},
+	{0xfefe, NULL,                                "SSL_RSA_FIPS_WITH_DES_CBC_SHA"},
+	{0xfeff, NULL,                                "SSL_RSA_FIPS_WITH_3DES_EDE_CBC_SHA"},
+	{0xff00, "GOST-MD5",                          "TLS_GOSTR341094_RSA_WITH_28147_CNT_MD5"},
+	{0xff01, "GOST-GOST94",                       "TLS_RSA_WITH_28147_CNT_GOST94"},
+	{0xff02, "GOST-GOST89MAC",                    NULL},
+	{0xff03, "GOST-GOST89STREAM",                 NULL},
+	{0xff85, "GOST2012256-GOST89-GOST89",         NULL},
+	{0xff87, "GOST2012256-NULL-STREEBOG256",      NULL},
+	{0xffe0, NULL,                                "SSL_RSA_FIPS_WITH_3DES_EDE_CBC_SHA"},
+	{0xffe1, NULL,                                "SSL_RSA_FIPS_WITH_DES_CBC_SHA"},
+	{0x10080, "RC4-MD5",                           "SSL_CK_RC4_128_WITH_MD5"},
+	{0x20080, "EXP-RC4-MD5",                       "SSL_CK_RC4_128_EXPORT40_WITH_MD5"},
+	{0x30080, "RC2-CBC-MD5",                       "SSL_CK_RC2_128_CBC_WITH_MD5"},
+	{0x40080, "EXP-RC2-CBC-MD5",                   "SSL_CK_RC2_128_CBC_EXPORT40_WITH_MD5"},
+	{0x50080, "IDEA-CBC-MD5",                      "SSL_CK_IDEA_128_CBC_WITH_MD5"},
+	{0x60040, "DES-CBC-MD5",                       "SSL_CK_DES_64_CBC_WITH_MD5"},
+	{0x700c0, "DES-CBC3-MD5",                      "SSL_CK_DES_192_EDE3_CBC_WITH_MD5"},
+	{0x80080, "RC4-64-MD5",                        "SSL_CK_RC4_64_WITH_MD5"},
+};
```

Comment:
This table is huge. It grows the data segment of the resulting shared library by roughly 35%, and overall library size increases by 3.3%. This might be acceptable, but I wonder whether we'll have to expand this table by more mappings for other TLS libraries, causing it to grow even bigger in size.

Reply:
I… didn't think about that 🤔. 35% is pretty big so I'm kinda meh on my own solution here.

As far as I can see, I don't think other mappings will be necessary (famous last words), as there are only 2 solutions in use :
- The OpenSSL/gnuTLS way, which goes through non-canonical names as a :-separated string, with various generic suite names (ie. "RC4"/"!RC4").
- The "enum"/"#define" with standard values (SecureTransport, mbedTLS). The API just accepts a `int *` to protocol IDs.

It might be possible to only build the needed mapping for each backend, which would somewhat lower the global size : mbedTLS/SecureTransport would be a simple `char *` to `int`, OpenSSL would still have a `char *` to `char *`.